src/Symfony/Component/Validator/Constraints/FileValidator.php

```diff
@@ -51,6 +52,10 @@ public function validate($value, Constraint $constraint)
             return;
         }
 
+        if ($value instanceof UploadedFileInterface) {
+            $value = new UploadedFile('', $value->getClientFilename(), $value->getClientMediaType(), $value->getSize(), $value->getError());
```

Comment:
passing an empty path looks wrong here. It will break things when retrieving the path to the file later in the validation logic to perform the assertions